compiler/rustc_builtin_macros/src/lib.rs

```diff
@@ -76,6 +77,8 @@ pub fn register_builtin_macros(resolver: &mut dyn ResolverExpand) {
         log_syntax: log_syntax::expand_log_syntax,
         module_path: source_util::expand_mod,
         option_env: env::expand_option_env,
+        core_panic_macro: panic::expand_panic,
+        std_panic_macro: panic::expand_panic,
```

Comment:
Looks like `core::panic` and `std::panic` ended up identical, so `std::panic` can simply reexport `core::panic`.

This also makes the `#[rustc_builtin_macro = "foo"]` form unnecessary, so it can be unsupported.

Reply:
They will only be identical in Rust 2021. For edition 2015 and 2018, it'll still be important to know whether to expand to the std or core version of the panic macro.

Reply:
Ah, I see, the token streams are produced with the same function, but they actually differ in hygiene because they use def-site hygiene and `core::panic` and `std::panic` are defined in different locations.

If `std::panic` reexported `core::panic`, then `$crate` produced by it would always refer to `core`.

Reply:
Yeah, exactly. It's very subtle. I'll add a comment about this.

Reply:
Could you actually add a comment about this to `fn expand_panic`, because it's not obvious.